src/CMakeLists.txt

```diff
@@ -70,6 +70,7 @@ include_directories(
     ${PROJECT_SOURCE_DIR}/third_party/cpp-btree
     ${PROJECT_SOURCE_DIR}/third_party/cqf/include
     ${PROJECT_SOURCE_DIR}/third_party/cxxopts/include
+    ${PROJECT_SOURCE_DIR}/third_party/magic_enum/include
```

Comment:
Hmm, can anybody help me here? Tried to do it the same way as `nlohmann_json` is included, but that does not work.

Reply:
Did you add the dependency to hyrise_impl (src/lib/CMakeLists.txt)?

Reply:
Something like 
```
diff --git a/src/lib/CMakeLists.txt b/src/lib/CMakeLists.txt
index 06786723..9dd975d8 100644
--- a/src/lib/CMakeLists.txt
+++ b/src/lib/CMakeLists.txt
@@ -703,6 +703,7 @@ set_source_files_properties(
 # -rdynamic tells the linker to export the library's symbols so that plugins can use them.
 target_link_libraries(hyrise_impl PUBLIC ${LIBRARIES} INTERFACE custom_jemalloc -rdynamic)
 target_link_libraries_system(hyrise_impl nlohmann_json::nlohmann_json)
+target_link_libraries_system(hyrise_impl magic_enum::magic_enum)
 target_include_directories(hyrise_impl PUBLIC ${CMAKE_BINARY_DIR})
```

Reply:
`target_link_libraries(hyrise_impl PUBLIC ${LIBRARIES} INTERFACE custom_jemalloc magic_enum -rdynamic)` works, but I thought this would be a system library (don't ask me why, just thought it's a lot like the json library looking where it will eventually be used).

Reply:
I am stupid. I think I slowly start to understand cmake a tiny bit.

Reply:
Have you tried adding magic_enum to the LIBRARIES variable?

Reply:
This only works if the hpp file is not within a sub directory (works for uninitialized_vector, not for others, I guess that's the reason why their include directories are also listed in `src/CMakeLists.txt`.
I am trying to find a way to make CMake search the sub directories as well, but this whole CMake thing is absolutely ...a

Reply:
Having the same issue with a different thing right now. Let's deal with that in #2032.